pkg/staticnetworkconfig/generator_test.go

```diff
@@ -416,17 +459,24 @@ var _ = Describe("StaticNetworkConfig.GenerateStaticNetworkConfigArchive", func(
 				FileContents: "static network config data of second host",
 			},
 		}
-		archiveBytes, err := GenerateStaticNetworkConfigArchive(data)
+		archiveBytes, err := snc.GenerateStaticNetworkConfigArchive(data)
 		Expect(err).ToNot(HaveOccurred())
 		Expect(archiveBytes).ToNot(BeNil())
 		checkArchiveString(archiveBytes.String(), data)
 	})
 })
 
-func checkArchiveString(archiveString string, allData []StaticNetworkConfigData) {
+func checkArchiveString(archiveString string, allData []snc.StaticNetworkConfigData) {
 	for _, data := range allData {
 		Expect(archiveString).To(ContainSubstring("tar"))
 		Expect(archiveString).To(ContainSubstring(filepath.Join("/etc/assisted/network", data.FilePath)))
 		Expect(archiveString).To(ContainSubstring(data.FileContents))
 	}
 }
+
+// escapeYAMLForJSON takes a YAML content string and escapes necessary characters to ensure it can be safely embedded within a JSON string.
```

Comment:
Why not use `json.Marshal` to do the job ? Example:
```suggestion
func escapeStringForJSON(s string) (string, error) {
	// Use json.Marshal to escape the string
	escaped, err := json.Marshal(s)
	if err != nil {
		return "", err
	}

	// json.Marshal returns a byte slice with double quotes around the string,
	// so we need to trim the double quotes
	return string(escaped[1 : len(escaped)-1]), nil
}
```

Reply:
@ori-amizur As we discussed, this PR was merged just before I came across this comment, so I'll incorporate this change in another PR.